plugins/sfIsaarPlugin/modules/sfIsaarPlugin/actions/actorEventsAction.class.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Actor - show event data as JSON
+ *
+ * @package    AccesstoMemory
+ * @subpackage Actor - initialize an editISAAR template for updating an actor
+ */
+class sfIsaarPluginActorEventsAction extends sfAction
+{
+  public function execute($request)
+  {
+    $actor = QubitActor::getBySlug($request->slug);
+
+    $criteria = new Criteria;
+    $criteria->add(QubitEvent::ACTOR_ID, $actor->id);
+
+    $data = array();
+    $data['total'] = count(QubitEvent::get($criteria));
+
+    $criteria->setOffset($request->skip);
+    $criteria->setLimit($request->limit);
+
+    $data['data'] = $this->assembleEventData($criteria);
+
+    return $this->renderText(json_encode($data));
```

Comment:
I think it would be good to set the "Content-type" header to 'application/json' too, e.g.:

```
$this->getResponse()->setHttpHeader('Content-type','application/json');
```

(Ref: https://ourcodeworld.com/articles/read/721/how-to-return-a-json-response-with-symfony-1-4)

Reply:
Good thinking. I tweaked the JS as well as it doesn't have to parse the response into JSON manually anymore (as jQuery does this automatically).

Reply:
👍